analysis/cpp/src/make-histograms-loop.cpp

```diff
@@ -60,33 +60,58 @@ int main(int argc, char* argv[]) {
     tree->Add(input_file.c_str());
     std::cout << "Loaded TTree \"" << tree_name << "\" with " << tree->GetEntries() << " events" << std::endl;
 
-    // primary vertex related variables
+    /*
+     Here we declare local variables that will be used to hold the data
+     for each of the variables we wish to read from the input
+     TTree file. There must be one locally declared variable for
+     each of the variables in the TTree that we wish to read.
+     */
+
+    // reco-level primary vertex related variables
     float pv_xpos = 0;
     float pv_ypos = 0;
     float pv_zpos = 0;
 
-    // variables for mc particles
-    int* mcpdg = new int[5000000]; // ROOT is dumb so just allocate a ton of memory for these arrays
-    float* mc_px = new float[5000000];
-    float* mc_py = new float[5000000];
-    float* mc_xpos = 0;
-    int n_mc_particles_in_event;
+    // truth-level primary vertex proxies (we take the vertex positions from the truth muons from the muon gun as a proxy)
+    float truth_pv_xpos = 0;
+    float truth_pv_ypos = 0;
+    float truth_pv_zpos = 0;
+
+    // mc particle related variables
+    const size_t mc_particle_array_size = 5000000;
+    int* mcpdg = new int[mc_particle_array_size]; // pdg id
+    int* mcgst = new int[mc_particle_array_size]; // MC generator status (1 == generated particle, 0 = decay ??)
```

Comment:
I've added the variable `mcgst` which is the "generator status" for each of the MC particles. Generally, in Monte Carlo simulations different particles are assigned different "status" identifiers, which indicate things like: "this particle was one of the original particles generated" or "this particle is not one of the original particles generated but is instead a particle that was produced as a result of a decay of another particle".

It seems that the ILCSoft assigns a generator status of `mcgst = 1` to the original produced particles. In our case, with the simple muon gun in which we generate events that are only 2 muons (a muon and an anti-muon), we expect only the muon and anti-muon to have `msgst = 1`.